app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java

```diff
@@ -599,20 +592,28 @@ public long folderSize(Context context) {
 
         switch (mode){
             case SFTP:
-                return SshClientUtils.execute(new SshClientSessionTemplate(path) {
+                return SshClientUtils.execute(new SshClientTemplate(path, false) {
                     @Override
-                    public Long execute(Session session) throws IOException {
-                        Session.Command cmd = session.exec(String.format("du -b -s \"%s\"",
-                                SshClientUtils.extractRemotePathFrom(path)));
+                    public Long execute(SSHClient client) throws IOException {
+                        Session session = client.startSession();
+                        String remotePath = SshClientUtils.extractRemotePathFrom(path);
+                        Session.Command cmd = session.exec(String.format("du -b -s \"%s\"", remotePath));
```

Comment:
Isn't there a method that does the same thing as the command?

Reply:
`net.schmizz.sshj.sftp.SFTPClient.size(path)`?
That seems to return predictable values like 4096, 20480 (or similar) for folders.

Reply:
Bear in mind that we are trying not to use commands because the responses are oriented towards users and can be unpredictable (in cases of error). But I don't know if this is true for SFTP.

Reply:
True indeed - hence #905 and this PR. Just not sure if it's fine by simply return the reported value by `SFTPClient.size(path)`.

Let's test the alternate method back at #905 and see how it goes.

Reply:
True indeed.

9ea93ed will use SFTPClient.size(path) alone. Copy is still functioning properly, but expect copy progress will report incorrect size.

You may want to refactor recursive copy's code so that each folder it descends to will update the total copy size to keep things beautiful.

![screenshot_1516030505](https://user-images.githubusercontent.com/251819/34950058-f50b7d74-fa4c-11e7-95e6-830548037d52.png)

Reply:
@TranceLove please test with #916 
I think with latest changes made, should fix all the problems with copy.

Reply:
Except the erroneous copy size.

Reply:
@VishalNehra yes, screenshot above is test with #916.

Reply:
@TranceLove @EmmanuelMess 

> `net.schmizz.sshj.sftp.SFTPClient.size(path)`?
> That seems to return predictable values like 4096, 20480 (or similar) for folders.

This is the size of *folder itself* — number of bytes, taken by *filenames* of it's direct descendants. I don't think, that there is any popular filesystem (on Linux or Windows), that can track total size of directory descendants. 

> Except the erroneous copy size. Maybe say: "x MB of unknown".

I second this. Trying to recursively compute directory size over network protocol have potential to take more time than copy itself (imagine copying deeply nested directory tree with thousands of individually small files — making full network roundtrip for each `stat()`).

Alternatively, if the remote directory is writable, Amaze can try to upload small script/executable (for example PowerShell program), to perform the job of `du`.

Reply:
> Alternatively, if the remote directory is writable, Amaze can try to upload small script/executable (for example PowerShell program), to perform the job of `du`.

Sorry, but allowing the app to inject programs/scripts to execute on remote servers is **always a bad idea**.

Reply:
@TranceLove you are already running `du` on remote server, aren't you?

Anyway, there is no need to create an executable — simply passing a script via PowerShell's `-Command` argument should do.

Reply:
> @TranceLove you are already running `du` on remote server, aren't you?

No denying. So I understand why I need to change this - though things maybe a bit safe in the UNIX world (relatively, compared to Windows).

Judging this, it is possible that the SFTP parts may no longer be able to report remaining spaces, used spaces... correctly because of the remote commands involved. :/